### PR TITLE
Update template files and dependencies

### LINKED ISSUE #259: Update template files and dependencies
We should update the template files with the new [generator-phovea v2](https://github.com/phovea/generator-phovea/releases/tag/v2.0.1) to update the dependencies and use Node.js 12.

### PR DESCRIPTION
**Pull request by thinkh**
Closes #259

### Summary

Update files using generator-phovea v2.0.1:

- use Node.js v12
- update CircleCI and gitlab config
- update to jQuery patch version
- update to D3 patch version
- update dev dependencies
- udpate webpack.config.js
- update buildInfo.js

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -65,40 +55,24 @@ const tsLoaderDev = [{
     options: {
       happyPackMode: true, // IMPORTANT! use happyPackMode mode to speed-up compilation and reduce errors reported to webpack,
       compilerOptions: {
-        "target": "es6",
-        "jsx": "react",
-        "jsxFactory": "h",
```
Comment (lehnerchristian) [inline webpack.config.js]: let's put this back

File: webpack.config.js [hunk 2]
```diff
@@ -315,7 +280,7 @@ function generateWebpack(options) {
   if (!options.bundle || options.isApp) {
     // extract the included css file to own file
     const p = new ExtractTextPlugin({
-      filename: (options.isApp || options.moduleBundle ? '[name]' : pkg.name) + (options.min && !options.nosuffix ? '.min' : '') + '.css',
```
Comment (lehnerchristian) [inline webpack.config.js]: [name] seems to be the newer version

File: package.json [hunk 3]
```diff
@@ -63,7 +63,7 @@
     "test:python": "test ! $(find tests -name \"*.py\") || python setup.py test",
     "predist:web": "npm run build:web && npm run docs:web",
     "dist:web": "mkdirp dist && cd build && tar cvzf ../dist/tdp_core.tar.gz *",
-    "dist:python": "python setup.py sdist bdist_wheel",
```
Comment (lehnerchristian) [inline package.json]: wheel is the newer one

**Comment by thinkh** [comment]
@lehnerchristian I addressed your feedback. Please review the changes again.

File: webpack.config.js (hunk 1 above)
Comment (thinkh) [reply webpack.config.js]: Fixed the different quote format.

File: package.json (hunk 3 above)
Comment (thinkh) [reply package.json]: See also https://github.com/phovea/generator-phovea/issues/259

File: webpack.config.js (hunk 2 above)
Comment (thinkh) [reply webpack.config.js]: See also https://github.com/phovea/generator-phovea/issues/257

**Pull request merged**